### DISCUSSION TITLE
Import device types access denied

### DISCUSSION THREAD

**Discussion by andershagman** [netbox-community/netbox#12788] [Q&A] [answered]
I get access denied when trying to import device types as a regular user with device type add, change, del and view rights.
This is the same on Netbox 3.3.6, 3.4.5 or demo.netbox.dev.
The same user can manually create a device type.
Same user can import devices, manufacturers or whatever.

From demo.netbox.dev:

![bild](https://github.com/netbox-community/netbox/assets/20284974/c722ca18-ea4b-42a1-b5ab-f0602e3e42a8)

**Comment by candlerb** [comment, marked as answer]
Which user on demo.netbox.dev - is it "andhag" ?  That has permission called "kalle" which is view,add (only) on device type.

According to the [source code](https://github.com/netbox-community/netbox/blob/v3.5.2/netbox/dcim/views.py#L1096-L1109), you need all the following permissions:

```
class DeviceTypeImportView(generic.BulkImportView):
    additional_permissions = [
        'dcim.add_devicetype',
        'dcim.add_consoleporttemplate',
        'dcim.add_consoleserverporttemplate',
        'dcim.add_powerporttemplate',
        'dcim.add_poweroutlettemplate',
        'dcim.add_interfacetemplate',
        'dcim.add_frontporttemplate',
        'dcim.add_rearporttemplate',
        'dcim.add_modulebaytemplate',
        'dcim.add_devicebaytemplate',
        'dcim.add_inventoryitemtemplate',
    ]
```

That's because importing the device type YAML can also create those related template objects.

**Reply by andershagman** [reply]
I have upgraded in steps from 2.8 to now in prod 3.3.6. The modulebay and inventoryitemtemplate are new and were not part of any predefined permissions. I created two new permission objects in this style:

![bild](https://github.com/netbox-community/netbox/assets/20284974/eb3a7a10-6aa0-4863-9951-90011a102f2e)

And now it works.

/Tanks
